crates/telio-proxy/src/proxy.rs

```diff
@@ -39,6 +40,48 @@ pub enum Error {
     /// Send failed
     #[error(transparent)]
     Send(#[from] SendTimeoutError<(PublicKey, DataMsg)>),
+    /// WG addr is null
+    #[error("Empty wg Addr")]
+    WgAddrNull,
```

Comment:
Are these new Errors WgAddrNull and RecvChanError necessary? If not, use ErrorKind::AddrNotAvailable/ErrorKind::BrokenPipe instead

Reply:
IMO `ErrorKind::AddrNotAvailable` and `ErrorKind::BrokenPipe` looks appropriate for those cases when the address (or ifc) is not available or when the channel is closed (broken).

> I also feel that reusing the ErrorKind errors would not propagate the error type properly.

Why do you think that?